### ISSUE TITLE
tarantool -e always enters interactive mode when stdin is a tty

### ISSUE THREAD

**Issue by un-def** [tarantool/tarantool#5040]
**Tarantool version:** 2.4.1-52-g4cd784999

**OS version:** Ubuntu 20.04 LTS

**Bug description:**

The comment of https://github.com/tarantool/tarantool/commit/a305d08ed9a097ddfe39faa3b7b1c4b5c4cc25c3 states that tarantool is “fully compatible with lua5.1” it terms of CLI. Actually, it's not quite true.

There is a difference in how `-e` parameter works. In PUC-Rio Lua and vanilla LuaJIT, if `-i` parameter is not passed, then the interpreter quits after execution of the expression regardless of whether or not stdin isatty.

In tarantool, the interpreter **always** enters interactive mode if stdin isatty regardless of the presence of `-i` parameter. The comment https://github.com/tarantool/tarantool/blob/03790ac5510648d1d9648bb2281857a7992d0593/src/main.cc#L752 says that the `-i` parameter actually **forces** interactive mode, not enables it (at least I could guess it from the comment).

The help message is misleading, it describes how it works in PUC-Rio Lua and vanilla LuaJIT:

```
  -e EXPR			execute string 'EXPR'
  -i				enter interactive mode after executing 'SCRIPT'
```

Actually, it is nearly a copy–paste of the original help message of original Lua REPL.

As I understand, the only way to enforce tarantool REPL to not to enter interactive mode is to disconnect its stdin from tty. For example, it works: `tarantool -e EXPR < /dev/null` or `echo | tarantool -e EXPR`.

The currect behavior breaks luarocks (vanilla luarocks, not luarocks that is shipped with tarantool) on Debian-based systems. luarocks [tries to find lua executable in PATH](https://github.com/luarocks/luarocks/blob/v3.3.1/src/luarocks/cmd.lua#L175) (and tarantool is linked to `/usr/bin/lua` via alternatives mechanism), [executes found binary](https://github.com/luarocks/luarocks/blob/v3.3.1/src/luarocks/util.lua#L439) with `-e` parameter and hangs on forever because tarantool does not exit.

**Steps to reproduce:**

* Execute in a terminal emulator: `tarantool -e 'print(_VERSION)'`.

**Expected behavior:** tarantool writes to stdout and exits.

**Actual behavior**:   tarantool writes to stdout and enters interactive mode.

**Comment by akudiyar** [comment]
Faced the same issue with Tarantool 2.3

**Comment by orchaton** [comment]
Same behavior with Tarantool 1.10.7 (latest).
If tarantool replaces default lua interpreter it must keep the same behavior.

**Comment by igormunkin** [comment]
It would be great to proxy `-j`/`-b` flags either for convenience.

**Comment by Totktonada** [comment]
> It would be great to proxy `-j`/`-b` flags either for convenience.

That's the issue regarding a bug. Can you file another one for passing more flags?

**Comment by igormunkin** [comment]
> Can you file another one for passing more flags?
Here you are: #5541.

**Comment by igormunkin** [comment]
> Can you file another one for passing more flags?

Here you are: #5541.

**Comment by Totktonada** [comment]
One more complain regarding `-i`: #5064.

**Comment by Buristan** [comment]
It would be nice to fix gh-5747 in scope of this issue too.

**Comment by artemreyt** [comment]
> It would be nice to fix #5747 in scope of this issue too.

This ticket is intended to solve problem with interactive mode.
Too highly specialized for your proposal.
